spec/tree_spec.rb

```diff
@@ -1,14 +1,79 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe Tree do
+  before(:all) do
+    @my_tree = Tree.new
+  end  
+
   it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+    expect(@my_tree.is_a? Tree).to be true
+  end
+
+  it 'should tell if the tree has any apples' do
+    expect(@my_tree.any_apples?).to eq(false)
+  end
+  
+  it 'should age one year on running the age command' do
+    @my_tree.age!
+    expect(@my_tree.age).to eq(1)
+  end
+
+  it 'should add apples on running the add_apples command' do
+    @my_tree.add_apples!
+    expect(@my_tree.any_apples?).to be true
+  end
+
+  it 'should return an apple on running the pick_an_apple command' do
+    expect(@my_tree.pick_an_apple.is_a? Apple).to be true
+  end
+
+  it 'should tell if the tree is dead' do
+    until @my_tree.age > 5
+      @my_tree.age!
+    end
+    expect(@my_tree.dead?).to be true
   end
+
 end
 
-describe 'Fruit' do
+describe Fruit do
+
+  before(:all) do
```

Comment:
@sjreich Thanks for the explanation. Currently, my tests depend on one another to succeed. For example, the test for adding apples must work before the successful completion of the test which picks apples.

`it 'should add apples on running the add_apples command' do
    @my_tree.add_apples!
    expect(@my_tree.any_apples?).to be true
  end

  it 'should return an apple on running the pick_an_apple command' do
    expect(@my_tree.pick_an_apple.is_a? Apple).to be true
  end`

If either dependent tests or using before(:all) is not considered a good practice, I can add a little more code to each of the tests so that a relevant tree operations are performed before testing the required functionality.

What do you think?